### ISSUE TITLE
Replace counts with percentages

### ISSUE THREAD

**Issue by Revadike** [bartervg/barter.vg#34]
Sure, counts work. They only work now, because barter isn't really growing that much. Once barter's userbase begins to fluctuate, we can no longer trust those collection counts when appraising our tradables.
Therefore, I vote to replace these tradable properties with percentages instead. Only show the detailed count when onclick or onhover. I feel like this would be a more reliable metric.

**Comment by Revadike** [comment]
I still think percentages are the way to go. Absolute numbers can be deceiving. Besides, I'd make it such that the absolute counts will be a popup on hover (e.g. with `title`).

**Comment by Luckz** [comment]
Seems your percentages are added, they just need to be based on "number of users with reasonably synced collections", for example for https://barter.vg/i/23562/ the total (3+8+2+30+29) is 72%.


Speaking about appraising, it's pretty hard to even get the absolute numbers on the graphs on https://barter.vg/bundle/3942/ (since you need to hover over the right bar with pixel precision). Maybe a meta-issue for getting to information more quickly?

**Comment by bartervg** [comment]
> Seems your percentages are added,

I think the request was for percentages on the offer page, but without details or screenshots, I'm speculating. I assumed this, but correct me if I'm wrong
![image](https://user-images.githubusercontent.com/28576132/115122362-7fe34680-9f85-11eb-9d1f-4060e0ffb754.png)


> the total (3+8+2+30+29) is 72%.

At first pass, it seems like it should add to 100%, but it's possible for everyone to have the game in tradable, wishlist (extra) and library, and thus this could add up to 300%. Unowned + owned (library, tradable, wishlist, and blacklist) doesn't add up to 100% due to `unowned` not meaning not in library, but also not filtered by offer preferences. Someone could exclude giveaways, and not appear in unowned despite not owning the game. You could argue that these users should appear in the (implicit) blacklist column then.

**Comment by Revadike** [comment]
Shouldn't it just be: 
A = pool of traders (I assume you're only looking at active registered barter.vg users)
B = subset of A, where item x is included in collection y, of all the trader in B

Then:
percentage = 100 * size(B) / size(A)

I don't get why it should be 300%. Looking at or combining other collections is messy. 

Also, regarding the whole issue. I think an option to choose between percentages or absolute number shown, will be the best of both worlds. And then, still have the secondary number as tooltip (using `title` attribute).

**Comment by bartervg** [comment]
On the offer page? You get best of both worlds on the item profile page already.

> percentage = 100 * size(B) / size(A)

Yes, this how it's calculated, where A is available active (3 days) users.

**Comment by Revadike** [comment]
> On the offer page?

 - offer page
 - browse page
 - collection page
 - match page (dropdown)

**Comment by Revadike** [comment]
> You get best of both worlds on the item profile page already.

Correct. And with what I suggested earlier, about making it an option. In that case, instead of showing both (more UI mess), either show absolute number (total count), or relative number (percentage), depending on the user preference. On hover, the user will be able to see the secondary number.

**Comment by bartervg** [comment]
What would be shown when there are more than 0 tradables, but a few enough so that the % is 0?

**Comment by Revadike** [comment]
I'd use 1, maybe 2 decimal points, or dynamically set it, depending on how small the percentage is.

**Comment by bartervg** [comment]
I don't see how 0.05% is preferable to 1

**Comment by Revadike** [comment]
Well, as I mentioned before, my main concern is the scalability of the site. Since the site probably keeps growing, and item with a wishlist count of 50 last year and 100 this year, falsely appears to be wanted more over time, but in reality the number just increase because of the growing userbase.

**Comment by bartervg** [comment]
Doesn't comparing the number of wishlists to the number of tradables mostly solve the problem of the changing size of userbase?

**Comment by Revadike** [comment]
Well, I for one, rarely check the number of tradables and just look at the number of wishlists. So no, it doesn't solve the problem.

**Comment by Revadike** [comment]
I mean, if that's the way you want to go, you should be displaying the ratio. That'd be fine too. The ratio indeed doesn't change with a growing userbase.

**Comment by bartervg** [comment]
50 tradables and 150 wishlists is different to me than 1 tradable and 3 wishlists. I think a lot is lost by reducing it to a ratio.

In terms of priority, preferences should be implemented first (and is long overdue), then an update of the offer page before spending time working on this.
